### PR TITLE
Admin UI for package locking

### PR DESCRIPTION
**Pull request by skofman1**
Screenshot: 
image

Yes, it's ugly on purpose 😆

### CODE REVIEW

**Comment by joelverhagen** [comment]
Screenshot please

File: src/NuGetGallery/Areas/Admin/Views/Home/Index.cshtml [hunk 1]
```diff
@@ -120,7 +120,7 @@
                 </p>
             </li>
         }
-	<li>
+	    <li>
```
Comment (scottbommarito) [inline src/NuGetGallery/Areas/Admin/Views/Home/Index.cshtml]: nit: spacing--remove the change to this line

File: tests/NuGetGallery.Facts/Areas/Admin/Services/ValidationAdminServiceFacts.cs [hunk 2]
```diff
@@ -40,7 +40,7 @@ public void ParsedToLines(string query, string[] lineArray)
                 var expected = lineArray.ToList();
 
                 // Act
-                var actual = ValidationAdminService.ParseQueryToLines(query);
+                var actual = Helpers.ParseQueryToLines(query);
```
Comment (scottbommarito) [inline tests/NuGetGallery.Facts/Areas/Admin/Services/ValidationAdminServiceFacts.cs]: Now that this is in its own class, we should move the testing to a separate file.

File: src/NuGetGallery/Areas/Admin/Views/LockPackage/Index.cshtml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            <div>
+                <h3>Results</h3>
+                <button type="button" onclick="selectall()">Select All</button>
+                <button type="button" onclick="deselectall()">Deselect All</button>
+                <table>
+                    <tbody>
+                        @for (int i = 0; i < Model.PackageLockStates.Count; i++)
+                        {
+                            <tr>
+                                <td>@Html.CheckBoxFor(model => model.PackageLockStates[i].IsLocked, new { @class = "islockedcheckbox" })</td>
```
Comment (scottbommarito) [inline src/NuGetGallery/Areas/Admin/Views/LockPackage/Index.cshtml]: Ideally I would structure the form in such a way that it is immediately clear that "this checkbox represents whether or not the package is locked or not". Right now there's just a checkbox next to each package ID which isn't very explicit. 

I think it'd be clearer if you put this into some sort of table, e.g.

| Id | Is Locked |
| --- | --- |
| Newtonsoft.Json | <ul> <li> [ ] </li> </ul> |

Comment (skofman1) [reply src/NuGetGallery/Areas/Admin/Views/LockPackage/Index.cshtml]: done

File: tests/NuGetGallery.Facts/Areas/Admin/Services/ValidationAdminServiceFacts.cs (hunk 2 above)
Comment (skofman1) [reply tests/NuGetGallery.Facts/Areas/Admin/Services/ValidationAdminServiceFacts.cs]: done

File: src/NuGetGallery/Areas/Admin/Helpers.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace NuGetGallery.Areas.Admin
+{
+    public static class Helpers
```
Comment (joelverhagen) [inline src/NuGetGallery/Areas/Admin/Helpers.cs]: Could we have a more specific name? `Helpers` can become a junk drawer easily.

File: src/NuGetGallery/Areas/Admin/Services/ValidationAdminService.cs [hunk 5]
```diff
@@ -78,35 +76,6 @@ public PackageDeletedStatus GetPackageDeletedStatus(int packageKey)
             return PackageDeletedStatus.NotDeleted;
         }
 
-        internal static IReadOnlyList<string> ParseQueryToLines(string query)
```
Comment (joelverhagen) [inline src/NuGetGallery/Areas/Admin/Services/ValidationAdminService.cs]: Thanks!

File: tests/NuGetGallery.Facts/Areas/Admin/Controllers/LockPackageControllerFacts.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                    new PackageLockState() { Id = "Test1", IsLocked = true },
+                    new PackageLockState() { Id = "Test5", IsLocked = true }
+                }
+            };
+
+            // Act
+            var result = await controller.Update(viewModel);
+
+            // Assert
+            Assert.Equal(2, packageRegistrationsInDb.Count(x => x.IsLocked));
```
Comment (chenriksson) [inline tests/NuGetGallery.Facts/Areas/Admin/Controllers/LockPackageControllerFacts.cs]: Test for unlocking a package?

**Pull request merged**